src/DotLiquid/Context.cs

```diff
@@ -169,7 +169,7 @@ public string HandleError(Exception ex)
                 return string.Empty;
 
             if (_errorsOutputMode == ErrorsOutputMode.Rethrow)
-                throw ex;
+                throw new Exception("Could not render the Liquid template.", ex);

```

Comment:
This is not the right fix - to preserve the stack the proper way is to just do a

throw;

in the original catch block.

This helper is only called there (Block::RenderAll). That code need to change to be told to rethrow in these circumstances.